concordances/handlers_test.go

```diff
@@ -0,0 +1,271 @@
+	expectedResponseCode: 503,
+	expectedContentType:  ContentTypeJson,
+	errorOp:              "retrieving",
+}
+
+var PUT_503 = TestCase{
+
+	description:          "PUT Service Not Available",
+	request:              newRequest("PUT", Path, GoodBody),
+	service:              &MockService{err: errors.New("")},
+	expectedResponseCode: 503,
+	expectedContentType:  ContentTypeJson,
+	errorOp:              "storing",
+}
+
+var DELETE_503 = TestCase{
+	description:          "DELETE Service Not Available",
+	request:              newRequest("DELETE", Path, GoodBody),
+	service:              &MockService{err: errors.New("")},
+	expectedResponseCode: 503,
+	expectedContentType:  ContentTypeJson,
+	errorOp:              "deleting",
+}
+
+var COUNT_503 = TestCase{description: "Service Not Available",
+	request:              newRequest("GET", "/concordances/__count", ""),
+	service:              &MockService{err: errors.New("")},
+	expectedResponseCode: 503,
+	expectedContentType:  ContentTypeJson,
+	errorOp:              "counting",
+}
+
+var GET_404 = TestCase{
+	description:          "GET Not Found",
+	request:              newRequest("GET", Path, ""),
+	service:              &MockService{model: db.Model{}},
+	expectedResponseCode: 404,
+	expectedContentType:  ContentTypeJson,
+}
+var DELETE_404 = TestCase{
+	description:          "Delete Not Found",
+	request:              newRequest("DELETE", Path, ""),
+	service:              &MockService{deleted: false},
+	expectedResponseCode: 404,
+	expectedContentType:  ContentTypeJson,
+}
+
+var DELETE_204 = TestCase{
+	description:          "204 Deleted",
+	request:              newRequest("DELETE", Path, ""),
+	service:              &MockService{deleted: true},
+	expectedResponseCode: 204,
+}
+
+var PUT_201 = TestCase{
+	description:          "PUT 201 Created",
+	request:              newRequest("PUT", Path, GoodBody),
+	service:              &MockService{created: true},
+	expectedResponseCode: 201,
+}
+
+var PUT_200 = TestCase{
+	description:          "PUT 200 Updated",
+	request:              newRequest("PUT", Path, GoodBody),
+	service:              &MockService{created: false},
+	expectedResponseCode: 200,
+}
+
+var GET_200 = TestCase{
+	description:          "GET 200 OK",
+	request:              newRequest("GET", Path, ""),
+	service:              &MockService{model: db.Model{UUID: TestConceptUuid, ConcordedIds: []string{"1", "2"}}},
+	expectedResponseCode: 200,
+	expectedResponseBody: GoodBody,
+}
+var COUNT_200 = TestCase{
+	description:          "COUNT 200 OK",
+	request:              newRequest("GET", "/concordances/__count", ""),
+	service:              &MockService{count: 0},
+	expectedResponseCode: 200,
+	expectedContentType:  "text/plain",
+	expectedResponseBody: "0",
+}
+
+func TestResponseCodesAndMessages(t *testing.T) {
+	testCases := []TestCase{GET_404, GET_503, GET_200, PUT_503, PUT_201, PUT_200, DELETE_503, DELETE_404, DELETE_204, COUNT_200, COUNT_503}
```

Comment:
Assigning each test case to a separate variable and then creating a list to loop through is rather verbose.  In addition, as this is the only place the tests are used then they should be kept in the scope of the test.  See [basic-tme-transformer](https://github.com/Financial-Times/basic-tme-transformer/blob/master/tme/handlers_test.go#L78) for an example of a more succinct version.